.ci/generate_documentation

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+import pathlib
+import subprocess
+import tempfile
+
+import git
+
+import gitutil
+from util import (
+    check_env,
+    ctx,
+    fail,
+    which,
+)
+
+
+sphinx_executable = which('sphinx-build')
+
+source_repo_dir = pathlib.Path(check_env('SOURCE_PATH')).resolve()
+source_dir = check_env('SOURCE_DIRECTORY')
+github_owner_and_name = check_env('SOURCE_GITHUB_REPO_OWNER_AND_NAME')
+_, repo_name = github_owner_and_name.split('/')
+
+context = ctx()
+config_factory = context.cfg_factory()
+gh_config = config_factory._cfg_element(cfg_type_name='github', cfg_name='github_com')
+
+with tempfile.TemporaryDirectory() as tmp_dir:
+    gitutil.clone_repository(
+        to_path=tmp_dir,
+        github_cfg=gh_config,
+        github_repo_path=github_owner_and_name,
```

Comment:
how about adding a new _optional_ argument "branch" to clone_repository (similar as git-clone offers it)? Sane default value might be '[refs/heads]/master'